packages/react-components/react-tree/src/components/TreeItem/TreeItem.types.ts

```diff
@@ -1,22 +1,40 @@
 import type { ComponentProps, ComponentState, Slot } from '@fluentui/react-utilities';
-import type { ARIAButtonElement, ARIAButtonElementIntersection, ARIAButtonSlotProps } from '@fluentui/react-aria';
+import {
+  BaseTreeItemElement,
+  BaseTreeItemElementIntersection,
+  BaseTreeItemProps,
+  BaseTreeItemSlots,
+} from '../BaseTreeItem/index';
 
-export type TreeItemElement = ARIAButtonElement<'div' | 'a'>;
+export type TreeItemElement = BaseTreeItemElement;
 
 /** @internal */
-export type TreeItemElementIntersection = ARIAButtonElementIntersection<'div' | 'a'>;
+export type TreeItemElementIntersection = BaseTreeItemElementIntersection;
 
-export type TreeItemSlots = {
+export type TreeItemSlots = BaseTreeItemSlots & {
   /**
-   * TreeItem root wraps around `props.content`
+   * Expand icon slot,
+   * by default renders a chevron icon to indicate opening and closing
    */
-  root: NonNullable<Slot<ARIAButtonSlotProps<'div' | 'a'>>>;
+  expandIcon?: Slot<'span'>;
+  /**
+   * Icon slot that renders right before main content
+   */
+  beforeIcon?: Slot<'span'>;
+  /**
+   * Icon slot that renders right after main content
+   */
+  afterIcon?: Slot<'span'>;
```

Comment:
is it possible that there is a before and after icon rendered at the same time? if not I would suggest to do something like `Button` does https://github.com/microsoft/fluentui/blob/b9ef9108ca7768db3fa086e7f601e8bdd37b8e90/packages/react-components/react-button/src/components/Button/Button.types.ts#L50

Reply:
I just sent a question to designers to be sure, but according to design yeah, it's possible to have both at the same time.

Reply:
alright if that's the case and both can be rendered at the same time, I would just suggest let's rename to `iconBefore` and `iconAfter`, `beforeIcon` sounds funny to me, but it's up to you 🤷.

Reply:
I renamed it for now to the iconAfter/iconBefore and let's see what the designers say 👍

Reply:
Ok, design team just confirmed it is possible to have both at the same time